### DISCUSSION TITLE
Profiler for performance problems in one kernel

### DISCUSSION THREAD

**Discussion by brian92617** [ROCm/ROCm#2543] [Q&A]
I have a single kernel that is taking up much of the GPU time.  Is there a profiler to help me understand what is going wrong in that kernel.  I'd like to know which instructions are taking up the time.  Basically, I'm looking for the AMD equivalent of NSight Compute.

**Comment by fluidnumericsJoe** [comment]
You might consider using omniperf ( https://amdresearch.github.io/omniperf/ )
